src/IECoreMaya/LiveScene.cpp

```diff
@@ -70,17 +70,71 @@
 #include "maya/MTransformationMatrix.h"
 #include "maya/MDagPathArray.h"
 #include "maya/MFnNurbsCurve.h"
+#include "maya/MGlobal.h"
 
 #include "OpenEXR/ImathBoxAlgo.h"
 
 #include <boost/algorithm/string.hpp>
 #include <boost/tokenizer.hpp>
+#include <boost/format.hpp>
+
+#include <maya/MFnSet.h>
 
 using namespace IECore;
 using namespace IECoreMaya;
 
 IE_CORE_DEFINERUNTIMETYPED( LiveScene );
 
+namespace
+{
+
+void readExportableSets( std::set<SceneInterface::Name> &exportableSets, const MDagPath &dagPath )
+{
+	// convert maya sets to tags
+	MSelectionList selectionList;
+	selectionList.add( dagPath );
+
+	MObjectArray sets;
+	if( !MGlobal::getAssociatedSets( selectionList, sets ) )
+	{
+		return;
+	}
+
+	for( unsigned int i = 0, numSets = sets.length(); i < numSets; ++i )
+	{
+		MStatus s;
+		MFnSet set( sets[i], &s );
+
+		if( !s )
+		{
+			continue;
+		}
+
+		MPlug exportPlug = set.findPlug( "ieExport", false, &s );
+
+		if( !s )
+		{
+			continue;
+		}
+
+		std::string exportName = boost::trim_copy( std::string( exportPlug.asString().asChar() ) );
```

Comment:
I'm not sure about this exportName prefix mechanism. Do we have a strong reason not to always just use the set name itself? Maybe we want the Maya user to be able to name their set whatever makes sense in a working scene but to strongly control the exported name? If that's the case, should we maybe consider it a full override on the tagName and not just a prefix? Or what we want is a prefix, should we call the plug "ieSetPrefix" instead? Also, do we need a separate "export me" toggle as well or is the existance of the attr enough?